packages/app-store/zohocalendar/lib/CalendarService.ts

```diff
@@ -0,0 +1,412 @@
+    }
+    try {
+      // needed to fetch etag
+      const existingEventResponse = await this.fetcher(`/calendars/${calendarId}/events/${uid}`);
+      const existingEventData = await this.handleData(existingEventResponse, this.log);
+
+      const query = stringify({
+        eventdata: JSON.stringify({
+          ...this.translateEvent(event),
+          etag: existingEventData.events[0].etag,
+        }),
+      });
+
+      const eventResponse = await this.fetcher(`/calendars/${calendarId}/events/${uid}?${query}`, {
+        method: "PUT",
+      });
+      eventRespData = await this.handleData(eventResponse, this.log);
+    } catch (error) {
+      this.log.error(error);
+      throw error;
+    }
+
+    try {
+      return {
+        ...eventRespData.events[0],
+        uid: eventRespData.events[0].uid as string,
+        id: eventRespData.events[0].uid as string,
+        type: "zoho_calendar",
+        password: "",
+        url: "",
+        additionalInfo: {},
+      };
+    } catch (error) {
+      this.log.error(error);
+      await this.deleteEvent(eventId, event);
+      throw error;
+    }
+  }
+
+  /**
+   * @param uid
+   * @param event
+   * @returns
+   */
+  async deleteEvent(uid: string, event: CalendarEvent, externalCalendarId?: string) {
+    const [mainHostDestinationCalendar] = event.destinationCalendar ?? [];
+    const calendarId = externalCalendarId || mainHostDestinationCalendar?.externalId;
+    if (!calendarId) {
+      this.log.error("no calendar id provided in deleteEvent");
+      throw new Error("no calendar id provided in deleteEvent");
+    }
+    try {
+      // needed to fetch etag
+      const existingEventResponse = await this.fetcher(`/calendars/${calendarId}/events/${uid}`);
+      const existingEventData = await this.handleData(existingEventResponse, this.log);
+
+      const response = await this.fetcher(`/calendars/${calendarId}/events/${uid}`, {
+        method: "DELETE",
+        headers: {
+          etag: existingEventData.events[0].etag,
+        },
+      });
+      await this.handleData(response, this.log);
+    } catch (error) {
+      this.log.error(error);
+      throw error;
+    }
+  }
+
+  private async getBusyData(dateFrom: string, dateTo: string, userEmail: string) {
+    const query = stringify({
+      sdate: dateFrom,
+      edate: dateTo,
+      ftype: "eventbased",
+      uemail: userEmail,
+    });
+
+    const response = await this.fetcher(`/calendars/freebusy?${query}`, {
+      method: "GET",
+    });
+
+    const data = await this.handleData(response, this.log);
+
+    if (data.fb_not_enabled || data.NODATA) return [];
+
+    return (
+      data.freebusy
+        .filter((freebusy: FreeBusy) => freebusy.fbtype === "busy")
+        .map((freebusy: FreeBusy) => ({
+          // using dayjs utc plugin because by default, dayjs parses and displays in local time, which causes a mismatch
+          start: dayjs.utc(freebusy.startTime, "YYYYMMDD[T]HHmmss[Z]").toISOString(),
```

Comment:
This throws error when I enable 'Show me busy during non-work hours option', reason being the date format sent by Zoho changes in that case.
![image](https://github.com/calcom/cal.com/assets/1780212/b5222e45-54b9-4513-9174-3729e455252d)
<img width="1117" alt="Screenshot 2023-09-25 at 1 01 22 PM" src="https://github.com/calcom/cal.com/assets/1780212/2d29ac1f-5687-403b-bdb9-1fcdd9c332a8">

Reply:
Also, I think it makes sense to document somewhere that one needs to enable this free busy option for events to be considered as busy.
![image](https://github.com/calcom/cal.com/assets/1780212/084c48be-4496-4cb6-967a-dd1bf34718ee)